src/System.Management.Automation/engine/regex.cs

```diff
@@ -233,33 +222,34 @@ internal static string Escape(string pattern, char[] charsNotToEscape)
             Span<char> temp = pattern.Length < StackAllocThreshold ? stackalloc char[pattern.Length * 2 + 1] : new char[pattern.Length * 2 + 1];
```

Comment:
This looks like a existing bug, the condition should probably have been `pattern.Length * 2 < StackAllocThreshold`.

I suggest refactoring to the recommended pattern (and in `Unescape` too):

```suggestion
            int lengthRequired = pattern.Length * 2 + 1;
            Span<char> temp = lengthRequired <= StackAllocThreshold ? stackalloc char[StackAllocThreshold] : new char[lengthRequired];
```